File: app/src/main/java/com/sample/android/contact/model/ContactPhoneNumber.java
package com.sample.android.contact.model;

public class ContactPhoneNumber {

    public String number;
    public String typeLabel;
    public int flagResId;

    public ContactPhoneNumber(String number, String typeLabel, int flagResId) {
        this.number = number;
        this.typeLabel = typeLabel;
        this.flagResId = flagResId;
    }

    @Override
    public boolean equals(Object o) {
        if (!(o instanceof ContactPhoneNumber)) {
            return false;
        }
        ContactPhoneNumber other = (ContactPhoneNumber) o;
        return other.number.replaceAll("\\s", "").equals(
                this.number.replaceAll("\\s", ""));
    }
}

File: app/src/main/java/com/sample/android/contact/domain/ContactPhoneNumber.kt
package com.sample.android.contact.domain

import android.os.Parcelable
import kotlinx.parcelize.Parcelize
import com.sample.android.contact.util.cleanString

@Parcelize
class ContactPhoneNumber @JvmOverloads constructor(
    val number: String,
    val typeLabel: String,
    var flagResId: Int? = null,
    var startPadding: Int? = null,
    var startMargin: Int? = null
) : Parcelable {

    override fun equals(other: Any?): Boolean {
        if (other !is ContactPhoneNumber) {
            return false
        }
        return other.number.cleanString() == number.cleanString()
    }

    override fun hashCode(): Int {
        return javaClass.hashCode()
    }
}
